### ISSUE TITLE
Type checker errors should be reported when syntax errors are present

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#17787]
Consider this case:
```fsharp
do
    let a = b.ToString()
```

<img width="203" alt="Screenshot 2024-09-25 at 11 37 57" src="https://github.com/user-attachments/assets/a797a97c-a694-459c-966d-2e19c59c906a">

The are two errors:
* syntax error: missing `in` part of the `let` expression
* unresolved `b` name

The second error is not reported. This has been like that for long time where we tried to minimize cascading type checker errors, so some of them are hidden when there are errors in preceding phases. However, with the significantly improved parser recovery it should be much better to report these errors now. Having any `in` expression helps in this case:
```fsharp
do
    let a = b.ToString()
    ()
```

<img width="207" alt="Screenshot 2024-09-25 at 11 38 03" src="https://github.com/user-attachments/assets/ae933ae5-31a7-45bd-a39f-efb4cf0b75a8">

But we want a better error reporting while editing too 🙂 

The same applies to other case where the parser recovery has been introduced, for example tuple expressions. When there are no syntax errors, type checker errors about the unresolved name and a missing overload are reported:
```fsharp
Math.Max(a)
```

<img width="200" alt="Screenshot 2024-09-25 at 11 41 06" src="https://github.com/user-attachments/assets/97bcaee6-124e-4758-861b-7195d78b5a3f">

When typing the second argument (and adding a syntax error in the tuple), the resolved names are reported, but the type checker errors are hidden (even though the parser recovery now allows the subsequent analysis):
```fsharp
Math.Max(a, )
```

<img width="208" alt="Screenshot 2024-09-25 at 11 41 44" src="https://github.com/user-attachments/assets/8fac2360-af1c-4481-8e2f-9030fec26744">

When the syntax error is fixed, everything is reported, even a warning about the unused expression result:
```fsharp
Math.Max(a, 1)
```

<img width="230" alt="Screenshot 2024-09-25 at 11 43 20" src="https://github.com/user-attachments/assets/20b06e94-88cf-4531-adec-91bdafbef076">

Typing subsequent arguments makes errors blink, which is not great:

<img width="248" alt="Screenshot 2024-09-25 at 11 45 06" src="https://github.com/user-attachments/assets/71bc16ef-ac4e-4eda-8a9a-65bc92705e12">

**Comment by abonie** [comment]
@auduchinok Do you know if it ever worked? Either way we should do it

**Comment by auduchinok** [comment]
> @auduchinok Do you know if it ever worked? Either way we should do it

No. Previously the parser would fail to a point where the analysis is not possible at all. Now the analysis is being done because of the improved parser, but type checker errors are hidden because of the syntax errors (that have been recovered). I suggest we should not hide these errors anymore.
